Prototype

muh200 opened the pull request:
At the moment, all this does is change the file type of video files in Google Cloud Storage. Users can interact with the server with the following command:

```shell
curl --data input_file=bucket_name/path/to/file --data output_file=bucket_name/path/to/file http://external-service-ip/ffmpeg
```

Commit:
Added Dockerfile that will build worker container

Commit:
Added commands to Dockerfile to build worker image

The Dockerfile now builds a worker image with ffmpeg and python
installed.
It uses the offical Docker image of python and installs ffmpeg from
Debian repositories since the default python image is based on Debian.

Commit:
Added worker's python file and requirements.txt

The requirements.txt will hold python dependencies needed to run the worker.

Commit:
Implemented /ffmpeg HTTP route

This runs ffmpeg locally. Eventually, it will take files from Google Cloud.

Commit:
Implemented /ffmpeg to download/upload files

The files are downloaded from Google Cloud Storage and processed locally
with ffmpeg. Then, the files are uploaded back to Google Cloud Storage.

isasi commented on worker/ffmpeg_worker.py:
```diff
@@ -0,0 +1,124 @@
+    This route changes the video container format by running the following:
+        ffmpeg -i {input_file} {output_file}
+
+    Args:
```
No need for Args: in docstring, if there are none

isasi commented on worker/ffmpeg_worker.py:
```diff
@@ -0,0 +1,124 @@
+    """Runs ffmpeg according to the request's specification.
+
+    The POST request supports the following arguments:
+        input_file: the input file in the Google Cloud Bucket specified
```
nit: Start with capital letter  - "The input file.."

You can just say "A GCS input file path "

isasi commented on worker/ffmpeg_worker.py:
```diff
@@ -0,0 +1,124 @@
+        None
+    """
+    client = storage.Client()
+    local_filename = _localize_filename(remote_filename)
```
remove this line as you are taking local_filename in the arguments.

isasi commented on worker/ffmpeg_worker.py:
```diff
@@ -0,0 +1,124 @@
+    return ffmpeg_logs
+
+
+def download_file(local_filename: str, remote_filename: str) -> None:
```
usual convention at google is that we put the read arguments first, followed by write arguments (altough you are not modifying the local_filename object, but the file referenced by that file path, the convention still makes sense)

isasi commented on worker/ffmpeg_worker.py:
```diff
@@ -0,0 +1,124 @@
+        None
+    """
+    client = storage.Client()
+    bucket_name, blob_name = _split_remote_filename(remote_filename)
```
no need for a separate function. just do 
bucket_name, blob_name =  os.path.normpath(remote_filename).split('/', 1) 
and let it throw an exception if there  is not "/" in the filename  .

isasi commented on worker/ffmpeg_worker.py:
```diff
@@ -0,0 +1,124 @@
+        A string with the ffmpeg logs.
+    """
+    remote_input_file = request.form['input_file']
+    local_input_file = _localize_filename(remote_input_file)
```
Use the functions in python tempfile library https://docs.python.org/3.5/library/tempfile.html to create tempfiles for input and output  instead of using  /tmp/*  .

isasi commented on worker/ffmpeg_worker.py:
```diff
@@ -0,0 +1,124 @@
+        stdout=subprocess.PIPE,
+        stderr=subprocess.STDOUT,
+        check=False).stdout
+    os.remove(local_input_file)
```
You can use with tempfile.NamedTemporaryfile(*delete=True) , so that you don't have to worry about deleting these explicitly.

Commit:
Reworded and fixed capitalization of comments

Commit:
Used tempfile library to create temporary files

The tempfile library creates temporary files that are automatically deleted.

isasi commented on worker/ffmpeg_worker.py:
```diff
@@ -38,52 +38,46 @@ def process_ffmpeg_command():
+        with tempfile.NamedTemporaryFile(suffix=output_format) as output_file:
+            download_file(remote_input_file, input_file.name)
+            ffmpeg_logs = subprocess.run(
+                ["ffmpeg", "-y", "-i", input_file.name, output_file.name],
```
"-y" option should be just before the output file option.

Commit:
Changed position of -y option

I also changed the double quotes to single quotes to stay consistent
with the rest of the file.
